Plugins/Flow.Launcher.Plugin.Explorer/Search/Everything/EverythingAPI.cs

```diff
@@ -172,5 +172,19 @@ private static void CheckAndThrowExceptionOnError()
                     throw new ArgumentOutOfRangeException();
             }
         }
+
+        public static async Task IncrementRunCounterAsync(string fileOrFolder)
+        {
+            try
+            {
+                await _semaphore.WaitAsync(TimeSpan.FromSeconds(1));
```

Comment:
This might causes issue. Better keep semaphore out of try. I remember the case that it doesn't wait but release, which causes exception.

Reply:
Can you explain a bit more? I fail to see what the problem is. If you have a SO link which explains the issue, give me that instead.

Reply:
https://stackoverflow.com/questions/24139084/semaphoreslim-waitasync-before-after-try-block

Reply:
Interesting. Thank you for the link. I've never had experience with async/await before and the points there make a lot of sense, in hindsight.

Reply:
fixed!